ls1intum/Artemis
# `Assessment`: Fix accepting complaints changes test case status

### Checklist
#### General
#### Server

### Motivation and Context
Closes #4334 (see Issue for more details)

### Description
The positive value was set based on points, with 0-point-feedback being non-positive. For test cases this is the wrong behavior, a test case is positive if its execution was successful, no matter the points.

When copying the feedback into the new result for the accepted complaint, the previous positive value now stays. Only if there is no positive-value set it will be calculated based on the points of the feedback.

### Steps for Testing
Follow the steps described in the issue.

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Complaints enabled

1. Setup the test case grading of the exercise to have 0-weight test cases (test cases that award 0 points when successful) 
2. Participate in the exercise (make sure that some of these 0-point tests are successful)
3. Submit a complaint as a student
4. Accept the complaint
5. Check the feedback view. The successful 0-point test cases should still be positive while failed test cases stayed negative.

### Review Progress

#### Code Review
#### Manual Tests

### Screenshots:
Old wrong behavior (failed 0 point test):
![grafik](https://user-images.githubusercontent.com/26540346/198827674-f93d6b17-04c9-4102-9bb4-48ad682a0035.png)

new (values stays):
